src/command_modules/azure-cli-sql/azure/cli/command_modules/sql/params.py

```diff
@@ -6,6 +6,49 @@
 from ._util import ParametersContext, patch_arg_make_required
 
 
+with ParametersContext(command='sql db') as c:
+    c.register_alias('database_name', ('--name', '-n'))
+
+with ParametersContext(command='sql db data-warehouse') as c:
+    c.register_alias('database_name', ('--database-name',))
+
+with ParametersContext(command='sql db replication-link') as c:
+    c.register_alias('database_name', ('--database-name',))
+
+with ParametersContext(command='sql db restore-point') as c:
+    c.register_alias('database_name', ('--database-name',))
+
+with ParametersContext(command='sql db service-tier-advisor') as c:
+    c.register_alias('database_name', ('--database-name',))
+
+with ParametersContext(command='sql db transparent-data-encryption') as c:
+    c.register_alias('database_name', ('--database-name',))
```

Comment:
As per my offline conversation with @tjprescott, we'll keep this as-is based on the design decision made earlier.